app/assets/javascripts/discourse/app/lib/to-markdown.js

```diff
@@ -48,6 +51,30 @@ export function clearBlockDecorateCallbacks() {
   blockDecorateCallbacks = [];
 }
 
+/**
+ * Allows to add support for custom text node transformations
+ * based on the next/previous elements.
+ *
+ * ```
+ * addTextDecorateCallback(function (text, nextElement, previousElement) {
+ *   if (
+ *     startRangeOpts &&
```

Comment:
If I'm just looking at the example, `startRangeOpts` is not defined anywhere which may be confusing.

Reply:
Oh yeah you are right, will simplify the example